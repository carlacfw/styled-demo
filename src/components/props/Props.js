import React, { Component } from 'react'
import styled, { keyframes } from 'styled-components'
import media from '../../shared/media'
import colours from '../../shared/colour'
import { ResponsiveWrapper } from '../../shared/components'

const VariableWapper = styled.div`
  display: flex;

  flex-direction: column;
  margin: 0;
  flex: 2;
  ${media.desktop`
    padding: 20px;
    width: 50%;
  `};
`
const SectionLable = styled.label`
  font-weight: bold;
  align-self: center;
`

const ColourButton = styled.button`
  background-color: ${props => props.background};
  color: ${props => props.color};
  margin: 1em;
  padding: 1em;
  border: 2px solid ${colours.secondary};
  border-radius: 6px;
  max-height: 45px;
  &:active {
    background-color: lightcyan;
    color: darkolivegreen;
  }
`

const ThickColourButton = ColourButton.extend`
  border: 10px solid blueviolet;
  max-height: 65px;
`

const BolderBiggerButton = styled.button.attrs()`
  color: ${props => (props.bolder ? 'black' : 'white')};
  font-weight: ${props => (props.bolder ? 'bolder' : 'normal')};
  height: ${props => (props.bigger ? '110' : '55')};
  width: ${props => (props.bigger ? '150px' : '100px')};
  background-color: lightblue;
  margin: 1em;
  padding: 1em;
  border: 2px solid ${colours.secondary};
  border-radius: 6px;
  &:active {
    background-color: lightcyan;
    color: darkolivegreen;
  }
  ${media.desktop`
  height: ${props => (props.bigger ? '200px' : '100px')};
  width: ${props => (props.bigger ? '300px' : '150px')};
  `};
`
const pulse = keyframes`
 0% {
    background-color: #001F5F;
  }
  100% {
    background-color: #FE4136;
  }
`

const PulsingBBB = BolderBiggerButton.extend`
  animation: ${pulse} 2s infinite;
`

export default class extends Component {
  render() {
    return (
      <VariableWapper onSubmit={this.handleSubmit}>
        <SectionLable>Passing props into styles</SectionLable>
        <ResponsiveWrapper>
          <ColourButton background="yellow" color="black">
            Yellow
          </ColourButton>
          <ColourButton background="blue" color="white">
            Blue
          </ColourButton>
          <ColourButton background="red" color="blue">
            Red
          </ColourButton>
        </ResponsiveWrapper>
        <SectionLable>Dynamic response to variables</SectionLable>
        <ResponsiveWrapper>
          <BolderBiggerButton
            bolder={this.props.match.isExact}
            bigger={this.props.match.isExact}
          >
            Exact
          </BolderBiggerButton>
          <BolderBiggerButton
            bolder={!this.props.match.isExact}
            bigger={!this.props.match.isExact}
          >
            Inexact
          </BolderBiggerButton>
        </ResponsiveWrapper>
        <SectionLable>Extending Styled Components</SectionLable>
        <ResponsiveWrapper>
          <ThickColourButton background="red" color="blue">
            Red
          </ThickColourButton>
          <PulsingBBB
            bolder={this.props.match.isExact}
            bigger={this.props.match.isExact}
          >
            Exact
          </PulsingBBB>
        </ResponsiveWrapper>
      </VariableWapper>
    )
  }
}
