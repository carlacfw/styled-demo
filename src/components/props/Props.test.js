import React, { Component } from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Props from './Props'
describe('The Props ', () => {
  it('It renders the Props correctly', () => {
    const wrapper = shallow(<Props match={{ isExact: true }} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
