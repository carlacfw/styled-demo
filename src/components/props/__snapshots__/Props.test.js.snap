// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Props  It renders the Props correctly 1`] = `
<styled.div>
  <styled.label>
    Passing props into styles
  </styled.label>
  <styled.div>
    <styled.button
      background="yellow"
      color="black"
    >
      Yellow
    </styled.button>
    <styled.button
      background="blue"
      color="white"
    >
      Blue
    </styled.button>
    <styled.button
      background="red"
      color="blue"
    >
      Red
    </styled.button>
  </styled.div>
  <styled.label>
    Dynamic response to variables
  </styled.label>
  <styled.div>
    <styled.button
      bigger={true}
      bolder={true}
    >
      Exact
    </styled.button>
    <styled.button
      bigger={false}
      bolder={false}
    >
      Inexact
    </styled.button>
  </styled.div>
  <styled.label>
    Extending Styled Components
  </styled.label>
  <styled.div>
    <styled.button
      background="red"
      color="blue"
    >
      Red
    </styled.button>
    <styled.button
      bigger={true}
      bolder={true}
    >
      Exact
    </styled.button>
  </styled.div>
</styled.div>
`;
